### DISCUSSION TITLE
Did You manage to succesefully install debian on Your buffalo device?

### DISCUSSION THREAD

**Discussion by DavidDah** [1000001101000/Debian_on_Buffalo#188] [Polls]
I would like to get rough idea how many people managed to successefully install debian on their buffalo devices.

**Comment by 1000001101000** [comment]
I’ve often wondered but haven’t done anything to try and track any of that. The lowest it could be is a few hundred given how many people have asked questions directly. I have no idea what portion of installs are from people I never hear from.

GitHub provides a few statistics but without history or the underlying details. From that I can see there are 200+ unique visitors in the two-week timeframe it shows. 

<img width="375" alt="image" src="https://github.com/1000001101000/Debian_on_Buffalo/assets/38921757/be155fe0-fa14-488c-95eb-f4e1dcd48e72">

**Comment by williamcfarmwise** [comment]
Yes, successfully installed debian on an old Terastation Live using the TS2Pro images install. I'd like to see if I can cross-compile the install images myself.

**Comment by bcoff12** [comment]
I've been routinely installing on LS210.  

I'm curious if anyone has a newer replacement single drive NAS they can install Debian on.  The 710 looks interesting (faster NIC).  I'd like to find something that will be available over the next 3 years or so.

**Comment by whcrg** [comment]
LS220D going strong for several years already. Though was also now almost a full year offline because moving and life happening etc.
Now online and should look at if bullseys is still current or if I should upgrade distros...

**Comment by eugrus** [comment]
Just installed Debian 12 onto LS-WXL.

Took more than an hour - reminded me the good old Windows 95 installation times back then ☺️

A more efficient approach probably would be to just have a ready system image to ``dd`` onto a disk and then just resize the main partition.

**Comment by 1000001101000** [comment]
Take a look at: https://github.com/1000001101000/Debian_on_Buffalo/wiki/Alternate-install-method-via-debootstrap-script

**Comment by ostroffjh** [comment]
Successfully installed on a Terrastation Pro a bit over two years ago, forgot all the details, as it was running fine until I needed to change its (static) IP address, and haven't been able access it since.  Not sure where to look for help with that.

**Reply by 1000001101000** [reply]
I’d probably set a PC’s ip address to a static one in the same subnet as the NAS and then connect it directly and try to log in that way

**Reply by ostroffjh** [reply]
I have a laptop at 192.168.1.19 (static.)  The NAS sSHOULD be at 192.168.1.110.  Both plugged into an unmanaged switch.  Ping gets me something like no route to host.  After I do "arp -s 192.168.1.110 nas-mac-address" pings just times out.  Earlier, I pulled the pair of disks with the root filesystem (raid1) and mounted them on another PC, and all relevant config files have 192.168.1.110.  The activity light on the switch is on, so the link is active. 
Note that the NAS has been running for two years with a different static IP (same subnet) and it disappeared after I changed the IP.  Of course at this point, I've forgotten exactly which file I edited under /etc.
Any other thoughts?

**Reply by 1000001101000** [reply]
It’s been a long time since I’ve done all that. Making sure the gateway is set the same matters to if I recall.

if you have filesystem access you could try just fixing the ip address etc.

**Reply by ostroffjh** [reply]
There's no gateway with a switch, but I had the same results using a router at 192.168.1.1.  I suppose I should just look at Debian support to confirm which files I need to check for the network configuration.  Is there any way (maybe a future enhancement) to get the NAS to display it's IP on the LCD panel?

**Reply by 1000001101000** [reply]
Network class was a long time ago, I feel like it uses the gateway ip + netmask to derive the broadcast address which is what’s important without the gateway present… it’s been a long time.

you can probably fire up wireshark and see the arp broadcasts and figure put the IP…. I think I even did that a few years ago with an iLo type thing.

if you’ve accessed the filesystem and read the interfaces file you should know all there is to know about what the NAS is configured to do.

**Reply by ostroffjh** [reply]
I actually did have wireshark going last week, and it didn't show anything coming back from the NAS.  That's why I'm wondering if I fat fingered one of the config files badly enough that it's not actually bringing up the network.  I've found some Debian network docs, so that will keep me busy for a while.  I'll report back eventually, but it may be a while.  "real life" (tm) calls.

**Comment by bcoff12** [comment]
I don't know if Debian is sensitive to whitespace issues in
      config files, but you could try wiping out the file and retyping
      it completely.
    Also triple check the values you have and interface name - here
      is a functional static IP I have on a debian machine:
    # The primary network interface - STATIC
      allow-hotplug enp2s0
      auto enp2s0
           iface enp2s0 inet static
           address 192.168.4.28/24
           gateway 192.168.4.1
    - Robert
    
    On 4/30/2024 8:29 PM, Jack Ostroff
      wrote:
    
    
      
      I actually did have wireshark going last week, and
        it didn't show anything coming back from the NAS. That's why I'm
        wondering if I fat fingered one of the config files badly enough
        that it's not actually bringing up the network. I've found some
        Debian network docs, so that will keep me busy for a while. I'll
        report back eventually, but it may be a while. "real life" (tm)
        calls.
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you commented.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/1000001101000/Debian_on_Buffalo/discussions/188#discussioncomment-9280875",
"url": "https://github.com/1000001101000/Debian_on_Buffalo/discussions/188#discussioncomment-9280875",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by ostroffjh** [reply]
Well, it's all working again now, and I simply have to plead temporary stupidity!  In looking more carefully at /etc/network/interfaces, first, I realized that I hadn't just swtiched IP addresses as I convinced myself, I had actually switched from dhcp to static IP.  Unfortunately, in process of commenting out the dhcp line, I also commented out the auto line.  I had added the static line and the lines specifying ip/gateway/netmask, but not the auto line.  So - the box simply was not even trying to enable the network.  The hardware was fine, so there was an active link (light on the router or switch) but the system was just ignoring anything coming in on that interface.  Once I added the "allow-hotplug eth0" (instead of auto) everything worked as expected.
Thanks for all the help and suggestions.
